src/common/common.ts

```diff
@@ -194,12 +194,36 @@ export const defaultCredentialTypes = [
 /**
  * A Verifiable Credential API configuration details.
  */
-export type VerifiableCredentialApiConfiguration = Partial<{
-  derivationService: UrlString;
-  issuerService: UrlString;
-  statusService: UrlString;
-  verifierService: UrlString;
-}>;
+export type VerifiableCredentialApiConfiguration =
+  // Legacy endpoints
+  Partial<{
+    derivationService: UrlString;
+    issuerService: UrlString;
+    statusService: UrlString;
+    verifierService: UrlString;
+  }> & {
+    // Spec-compliant endpoints, available in the `specCompliant` object
+    specCompliant: Partial<{
+      derivationService: UrlString;
+      issuerService: UrlString;
+      issuerCredentialAll: UrlString;
+      holderPresentationAll: UrlString;
+      statusService: UrlString;
+      credentialVerifierService: UrlString;
+      presentationVerifierService: UrlString;
+      queryService: UrlString;
+      exchangeService: UrlString;
+      proveService: UrlString;
+    }>;
+  } & {
+    // Legacy endpoints, available in the `legacy` object too to ease transition
+    legacy: Partial<{
+      derivationService: UrlString;
+      issuerService: UrlString;
+      statusService: UrlString;
+      verifierService: UrlString;
+    }>;
+  };
```

Comment:
Should this be a union? i.e., it's either spec-compliant or it's legacy.

Reply:
A given issuer could be both, as legacy and spec-compliant endpoints aren't mutually exclusive. As a matter of fact, we rely on this for the 2.1 release being non-breaking.

Reply:
Ah! Okay!